### PR TITLE
Expose pigpio library microsecond tick value plus diff utility method

### PR DESCRIPTION
**Pull request by erikma**
- Exposes getTick() and tickDiff() at the Node.js pigpio module level since these methods are not specific to a single GPIO pin in the GPIO class.
- Added spin-wait UT that utilizes the new methods.
- For issue #58

### CODE REVIEW

**Comment by erikma** [comment]
Recommend this PR plus #64 Waves as pigpio NPM package version 1.2.0

File: README.md [hunk 1]
```diff
@@ -312,5 +316,6 @@ Here are a few links to other hardware specific Node.js packages that may be of
 - [i2c-bus](https://github.com/fivdi/i2c-bus) - I2C serial bus access
 - [spi-device](https://github.com/fivdi/spi-device) - SPI serial bus access
 - [mcp-spi-adc](https://github.com/fivdi/mcp-spi-adc) - Analog to digital conversion with the MCP3002/4/8, MCP3202/4/8 and MCP3304
+- [pigpio-dht](https://github.com/depuits/pigpio-dht) - Implements logic to read DHT11 or DHT22/AM2302 temperature and relative humidity sensor
```
Comment (fivdi) [inline README.md]: Have you verified that pigpio-dht functions correctly?

File: doc/global.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+## Global
+
+The global functions are defined at the pigpio module level.
+
+#### Functions
+  - [getTick()](#getTick)
+  - [tickDiff(startTick, endTick)](#tickDiff)
```
Comment (fivdi) [inline doc/global.md]: the tickDiff link doesn't work. Replacing `#tickDiff` with `#tickdiffstarttick-endtick` should fix it.

File: doc/global.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+Gets the current unsigned 32-bit integer value of the number of microseconds
+since system boot. This value wraps around the 32-bit space in just over an hour.
+Use [tickDiff()](#tickDiff) to get the difference between two tick values, to
+ensure the correct JavaScript operations are used to account for the possibility
+of overflow.
+
+```js
+const pigpio = require('pigpio');
+
+let startUsec = gpio.getTick();
```
Comment (fivdi) [inline doc/global.md]: `gpio` is undefined and should be `pigpio`

File: doc/global.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+// Do some time-consuming things.
+
+let currentUsec = gpio.getTick();
```
Comment (fivdi) [inline doc/global.md]: `gpio` is undefined and should be `pigpio`

File: doc/global.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+let deltaUsec = gpio.tickDiff(startUsec, currentUsec);
```
Comment (fivdi) [inline doc/global.md]: `gpio` is undefined and should be `pigpio`

File: doc/global.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+#### tickDiff(startTick, endTick)
+Returns the difference in microseconds between the end and start tick counts.
+The tick counts can be retrieved using (getTick())[#getTick], or may be passed
+in a GPIO event callback.
+
+```js
+const pigpio = require('pigpio');
+
+let startUsec = gpio.getTick();
```
Comment (fivdi) [inline doc/global.md]: `gpio` is undefined and should be `pigpio`

File: doc/global.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+// Do some time-consuming things.
+
+let currentUsec = gpio.getTick();
```
Comment (fivdi) [inline doc/global.md]: `gpio` is undefined and should be `pigpio`

File: doc/global.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+let deltaUsec = gpio.tickDiff(startUsec, currentUsec);
```
Comment (fivdi) [inline doc/global.md]: `gpio` is undefined and should be `pigpio`

File: pigpio.js [hunk 9]
```diff
@@ -320,4 +332,3 @@ module.exports.configureSocketPort = (port) => {
 
 module.exports.CLOCK_PWM = 0; // PI_CLOCK_PWM;
 module.exports.CLOCK_PCM = 1; // PI_CLOCK_PCM;
-
```
Comment (fivdi) [inline pigpio.js]: Don't remove the empty line at the end of pigpio.js.

File: README.md (hunk 1 above)
Comment (erikma) [reply README.md]: Amusingly enough I have a branch in a fork of it, https://github.com/erikma/pigpio-dht/tree/dev/erikma/fixRepeatedReadings, that fixes the 2nd and later readings. If all you want is one reading per node.js invocation it's OK, if like me you want to run readings every 2 minutes forever, you need my change which is what prompted me to issue this PR for pigpio.

File: doc/global.md (hunk 2 above)
Comment (erikma) [reply doc/global.md]: Thx, not sure how to test these links and was guessing from other .md files

File: README.md (hunk 1 above)
Comment (erikma) [reply README.md]: My intent for that fork is to first get a new pigpio npm version on npmjs, then update my fork and issue a PR against -dht using these changes.

**Comment by fivdi** [comment]
@erikma Thank you very much.

**Pull request merged**